google/cloud/internal/big_endian.h

```diff
@@ -40,14 +40,14 @@ std::string EncodeBigEndian(T value) {
   static_assert(std::numeric_limits<unsigned char>::digits == 8,
                 "This code assumes an 8-bit char");
   using unsigned_type = typename std::make_unsigned<T>::type;
-  unsigned_type const n = value;
+  unsigned_type const n = *reinterpret_cast<unsigned_type*>(&value);
```

Comment:
This one (potentially signed -> unsigned) was fine methinks.

Reply:
> actually, I think the reinterpret_cast is needed here. Left it in.

Yes ... sorry.  While the (signed -> unsigned) conversion is well-defined, it is important that this does not change the representation, so reinterpretation is needed.